src/Thinktecture.Relay.Server/Transport/InMemoryServerHandler.cs

```diff
@@ -10,8 +10,24 @@ internal class InMemoryServerHandler<TResponse> : IServerHandler<TResponse>
 		public event AsyncEventHandler<TResponse> ResponseReceived;
 		public event AsyncEventHandler<IAcknowledgeRequest> AcknowledgeReceived;
 
-		public async Task DispatchResponseAsync(TResponse response) => await ResponseReceived.InvokeAsync(this, response);
+		public InMemoryServerHandler(IServerDispatcher<TResponse> serverDispatcher)
+		{
+			switch (serverDispatcher)
```

Comment:
Ggf. pattern matching nehmen statt switch?
`if (serverDispatcher==null) throw`
und dann `if (serverDispatcher is InMemoryDispatcher<TResponse> dispatcher) { dispatcher.doSomething(); } else { throw }`